src/coord/src/catalog.rs

```diff
@@ -2502,10 +2548,52 @@ enum SerializedCatalogItem {
         create_sql: String,
         // The name "eval_env" is historical.
         eval_env: Option<SerializedPlanContext>,
-        persist_name: Option<String>,
+        // Previous versions used "persist_name" as the field name here.
+        #[serde(alias = "persist_name")]
+        table_persist_name: Option<String>,
+        source_persist_details: Option<SerializedSourcePersistDetails>,
     },
 }
 
+/// Serialized source persistence details. See `SourcePersistDesc` for an explanation of the
+/// fields.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct SerializedSourcePersistDetails {
```

Comment:
fyi: https://serde.rs/json.html

Reply:
Yep, it should work for us, and I added `alias = "persist_name"` to the renamed field in `SerializedCatalogItem`.